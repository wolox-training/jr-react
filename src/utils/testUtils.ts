import { screen, fireEvent } from '@testing-library/react';

export const setInputByLabelText = (name: string, value: string) => {
  fireEvent.input(screen.getByLabelText(name), {
    target: {
      value
    }
  });
};
