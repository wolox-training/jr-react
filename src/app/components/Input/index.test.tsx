import React from 'react';
import { render, screen } from '@testing-library/react';

import { setInputByLabelText } from '~utils/testUtils';

import InputComponent from './index';

describe('Component Alert', () => {
  it('Renders successfully', () => {
    const { asFragment } = render(
      <InputComponent labelText="Nombre" inputRef={input => input} name="firstName" />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('If an error message is sent, it is displayed on the screen', () => {
    render(
      <InputComponent labelText="Email" inputRef={input => input} name="email" errorMessage="invalid email" />
    );
    expect(screen.getByRole('alert')).toHaveTextContent('invalid email');
  });

  it('the input allows to be edited by the user', () => {
    render(
      <InputComponent labelText="Email" inputRef={input => input} name="email" errorMessage="invalid email" />
    );
    setInputByLabelText('Email', 'example@gmail.com');
    expect(screen.getByLabelText('Email')).toHaveValue('example@gmail.com');
  });
});
