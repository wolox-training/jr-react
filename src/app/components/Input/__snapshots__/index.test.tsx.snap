// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Alert Renders successfully 1`] = `
<DocumentFragment>
  <div
    class="row wrap contentInput"
  >
    <label
      class="label"
      for="firstName"
    >
      Nombre
    </label>
    <input
      class="full-width input"
      id="firstName"
      name="firstName"
      type="text"
    />
  </div>
</DocumentFragment>
`;
